### PR TITLE
Parsing TLC output when traces are of length 1

### PR DESCRIPTION
**Pull request by ivan-gavran**
Fix for the issue #55

### CODE REVIEW

File: modelator_py/apalache/cli.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def pure(self):
             self._stdin is not None
         ), "The pure interface requires json input in stdin"
         json_dict = stdjson.loads(self._stdin.read())
-
+
```
Comment (danwt) [inline modelator_py/apalache/cli.py:27]: Did you run the hooks? I think the formatter would not have allowed this whitespace.

Comment (danwt) [inline modelator_py/apalache/cli.py:27]: Did you run the hooks? I think the formatter would not have allowed this whitespace.

File: modelator_py/apalache/pure.py [hunk 2]
```diff
@@ -95,22 +95,25 @@ def filename(full_filename):
...
     Returns an ExecutionResult with .process and .files properties. Contains the
     subprocess result, and the list of filesystem files (and contents).
     """
+    
     assert not (cmd is not None and json is not None)
-    assert (cmd is not None) or (json is not None)
-
-    if json is not None:
+    assert (cmd is not None) or (json is not None)    
+    if json is not None:
```
Comment (danwt) [inline modelator_py/apalache/pure.py:108]: nice!

File: modelator_py/util/tlc/stdout_to_informal_trace_format.py [hunk 3]
```diff
@@ -15,41 +15,59 @@ def trace_lines_model_checking_mode(stdout) -> typing.List[typing.List[str]]:
...
             and (not line.startswith("Progress"))
         ) or ("Model checking completed" in line)
 
+        single_state_footer = "Finished in" in line
+
+        return single_state_footer or multi_state_footer
+
+    def format_single_state_trace(trace):
+        # we need this because for single-state trace errors ther will be no states
+        # enumerated in the output (but we need them in postprocessing)
```
Comment (danwt) [inline modelator_py/util/tlc/stdout_to_informal_trace_format.py:44]: If possible, it might be better to change the postprocessing. But you're the judge.

**Review by danwt — APPROVED** [review]
It looks nice, I like the style. I didn't mentally check the logic, but I can on Monday if you want me to. If you're confident it works then please go ahead and merge after addressing the points about the commit hooks and the other test case. 

Nice work overall!

**Comment by danwt** [comment]
Will it work for the case that both

1. the initial state is a violation
2. it is run using `-continue`

are simultaneously true?

File: modelator_py/apalache/cli.py (hunk 1 above)
Comment (ivan-gavran) [reply modelator_py/apalache/cli.py:27]: I added hooks only today (so I guess it didn't pick up this). WIll fix it.

Comment (danwt) [reply modelator_py/apalache/cli.py:27]: Ah you mean github didn't pick up? Did you add extra hooks? There are these hooks already https://github.com/informalsystems/modelator-py/blob/main/.pre-commit-config.yaml

Comment (ivan-gavran) [reply modelator_py/apalache/cli.py]: I was not precise: I did not have pre-commit installed locally. Once I did install it, it was only working for future commits (I had to run `pre-commit run --all-files` to check previous commits, which I did now)

File: modelator_py/util/tlc/stdout_to_informal_trace_format.py (hunk 3 above)
Comment (ivan-gavran) [reply modelator_py/util/tlc/stdout_to_informal_trace_format.py]: done

**Comment by ivan-gavran** [comment]
> Will it work for the case that both
> 
> 1. the initial state is a violation
> 2. it is run using `-continue`
> 
> are simultaneously true?

Thanks for that! Now it works (didn't before).

**Comment by ivan-gavran** [comment]
Overall, thanks for your review, @danwt, it is extremely helpful that you are catching corner cases and pushing for doing things properly . You managed to make this PR much better :)

**Pull request merged**